Jenkinsfile.github

```diff
@@ -25,117 +25,139 @@
  */
 @Library('csm-shared-library') _
 
-def pythonImage = 'artifactory.algol60.net/csm-docker/stable/csm-docker-sle-python'
-def pythonVersion = '3.10'
-def major
-def minor
-def patch
-def isStable = env.TAG_NAME != null ? true : false
-if ( isStable ) {
-    (major, minor, patch) = env.TAG_NAME.tokenize('.')
-    major = major.replaceAll("^v","")
+def pythonImage = 'artifactory.algol60.net/csm-docker/stable/csm-docker-sle-python:3.10'
+
+// Disable pr-merge builds; not used.
+if (env.BRANCH_NAME ==~ ~"^PR-\\d+") {
+    currentBuild.result = 'SUCCESS'
+    echo "Pull-Request builds are skipped."
+    return
+} else if (env.BRANCH_NAME ==~ ~"^dependabot/github_actions") {
+    currentBuild.result = 'SUCCESS'
+    echo "Dependabot GitHub action builds are skipped."
+    return
 }
+
+// Only consider X.Y.Z and X.Y.Z.postN tags as stable.
+// Never consider X.Y.Z{[a|b|rc} or X.Y.Z.* tags (that are not post-releases) as stable.
+// The ==~ operator performs an exact match.
```

Comment:
Is this comment needed? If so, I don't see ==~ being used below.

Reply:
I copied it from another build, let me fix.

Reply:
Used two lines below this one.

Reply:
` 45 def isStable = (env.TAG_NAME != null & env.TAG_NAME ==~ stableToken) ? true : false`